resources/ts/controllers/input/cash_counter_controller.ts

```diff
@@ -0,0 +1,203 @@
+import { type } from 'jquery';
+import { Builder } from '../../ModalFormBuilder/builder';
+import { ApplicationController } from '../ApplicationController';
+import { InsertEvent } from '../modal_theatre_controller';
+
+interface ChipType {
+    id: number,
+    color: string,
+    denomination: number,
+    amount: number,
+}
+
+export default class extends ApplicationController {
+
+    static values = {
+        chipAmounts: String
+    };
+
+    static targets = [
+        'userInput', 'total', 'output'
+    ];
+
+
+    declare modalTheatre: HTMLDivElement;
+    declare chipTypes: ChipType[];
+    declare readonly userInputTarget: HTMLInputElement;
+    declare readonly totalTarget: HTMLSpanElement;
+    declare readonly outputTarget: HTMLInputElement;
+    declare readonly chipAmountsValue: string;
+
+    connect() {
+        //@ts-ignore
+        this.modalTheatre = document.getElementById('modal-theatre');
+        this.chipTypes = [];
+        let obj = JSON.parse(this.chipAmountsValue);
+        let keys = Object.keys(obj);
+
+        for(let i = 0; i < keys.length; i++) {
+            let key = keys[i];
+            this.chipTypes.push({
+                id: parseInt(key),
+                color: obj[key].color,
+                denomination: parseFloat(obj[key].denomination),
+                amount: parseInt(obj[key].amount),
+            });
+        }
+
+        let total = 0;
+
+        this.chipTypes.forEach(type => {
+            total += type.denomination * type.amount;
+        });
+
+        this.totalTarget.innerText = `(${total})`;
+
+        this.bindEventListeners();
+    }
+
+    calculateChipCut(event: Event) {
```

Comment:
This function needs to be split up.

We need to keep only the parts that generate chipsToHandout, Then we need to move everything else that togles the modal to another function.